### PR TITLE
[Java] Add release notes for 4.25.0

### PR DESCRIPTION
**Pull request by cschubertcs**
## What Has Changed?

This PR adds release not for our 4.25.0 release.

### CODE REVIEW

File: docs-java/release-notes/release-notes-15-to-29.mdx [hunk 1]
```diff
@@ -1,3 +1,16 @@
+# 4.25.0 - October 13, 2023
+
+## Improvements
```
Comment (CharlesDuboisSAP) [inline docs-java/release-notes/release-notes-15-to-29.mdx:3]:
```suggestion
## 4.25.0 - October 13, 2023

### Improvements
```

File: docs-java/release-notes/release-notes-15-to-29.mdx [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+
+- Dependency Updates:
+  - SAP dependency updates:
+    - Update `com.sap.cds.repackaged:odata-v4-lib` used in the `master-data-integration` modules from `2.2.0` to `2.3.0`
+  - Other dependency updates:
+    - Minor version updates:
+      - Update `com.google.api:api-common` used in the `blockchain-client-fabric` module from `2.17.0` to `2.18.0`
+      - Update `org.checkerframework:checker-qual` declared for dependency convergence from `3.38.0` to `3.39.0`
+      - Update `commons-io:commons-io` from `2.13.0` to `2.14.0`
```
Comment (CharlesDuboisSAP) [inline docs-java/release-notes/release-notes-15-to-29.mdx:12]: No links?

**Review by CharlesDuboisSAP — CHANGES_REQUESTED** [review]

File: docs-java/release-notes/release-notes-15-to-29.mdx (hunk 2 above)
Comment (cschubertcs) [reply docs-java/release-notes/release-notes-15-to-29.mdx]: I think it's not a requirement and I saw that we did it rather irregularly.
But true, that makes it easier for our readers.
Added!

File: docs-java/release-notes/release-notes-15-to-29.mdx [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+    - Update [`com.sap.cds.repackaged:odata-v4-lib`](https://central.sonatype.com/search?q=g:com.sap.cds.repackaged%20%20a:odata-v4-lib) used in the `master-data-integration` modules from `2.2.0` to `2.3.0`
   - Other dependency updates:
     - Minor version updates:
-      - Update `com.google.api:api-common` used in the `blockchain-client-fabric` module from `2.17.0` to `2.18.0`
-      - Update `org.checkerframework:checker-qual` declared for dependency convergence from `3.38.0` to `3.39.0`
-      - Update `commons-io:commons-io` from `2.13.0` to `2.14.0`
+      - Update [`com.google.api:api-common`](https://search.maven.org/search?q=g:com.google.api%20AND%20a:api-common) used in the `blockchain-client-fabric` module from `2.17.0` to `2.18.0`
+      - Update [`org.checkerframework:checker-qual`](https://central.sonatype.com/search?q=g:org.checkerframework%2520%20a:checker-qual) declared for dependency convergence from `3.38.0` to `3.39.0`
+      - Update [`commons-io:commons-io`](https://central.sonatype.com/search?q=g:commons-io%20%20a:commons-io) from `2.13.0` to `2.14.0`
```
Comment (CharlesDuboisSAP) [inline docs-java/release-notes/release-notes-15-to-29.mdx:12]:
```suggestion
      - Update [`org.checkerframework:checker-qual`](https://central.sonatype.com/search?q=g:org.checkerframework+a:checker-qual) declared for dependency convergence from `3.38.0` to `3.39.0`
      - Update [`commons-io:commons-io`](https://central.sonatype.com/search?q=g:commons-io%20a:commons-io) from `2.13.0` to `2.14.0`
```

**Review by CharlesDuboisSAP — APPROVED** [review]

**Pull request merged**